Client Database System/Client Database System/Product and Services.cpp

```diff
@@ -8,13 +8,74 @@
 
 #include "Product and Services.hpp"
 #include <iostream>
+#include <fstream>
+#include <iomanip>
+#include <string>
 #include <vector>
 
 using namespace std;
 
-void AddPrice(vector<int>& priceVector){
+Products_and_Services::Products_and_Services(){
+    product = "No product";
+    serviceDescription = "No service description";
+}
+
+void Product_and_Services::AddPrice(vector<int>& priceVector){
     int userPrice;
     
     cin >> userPrice;
     priceVector.push_back(userPrice);
 }
+
+void Product_and_Services::ReadFromFile(vector<string>& productVector, vector<string>& descriptionVector){
+}
+
+void Product_and_Services::ListServices(){
+}
+
+void Product_and_Services::AddNewService(string product, string serviceDescription, vector<string>& productVector, vector<string>& descriptionVector){
+    cout << "Input product, service description: ";
+    cin >> product;
+    cin >> serviceDescription;
+    
+    productVector.push_back(product);
+    descriptionVector.push_back(serviceDescription);
+}
+
+void Product_and_Services::UpdateService(string product, string serviceDescription, vector<string>& productVector, vector<string>& descriptionVector){
+    string userProduct;
+    string userService;
+    
+    cin >> userProduct;
+    cin >> userService;
+    
+    unsigned int i;
+    for(i = 0; i < productVector.size(); ++i){
+        if(userProduct == productVector.at(i)){
+            cout << "Input new product: " << endl;
+            cin >> userProduct;
+        }
+    }
+    for(i = 0; i < descriptionVector.size(); ++i){
+        if(userService == descriptionVector.at(i)){
+            cout << "Input new service: " << endl;
+            cin >> userService;
+        }
+    }
+}
+
+void Product_and_Services::PrintServicesReport(vector<string>& productVector, vector<string>& descriptionVector){
+    unsigned int i;
+    
+    for (i = 0; i < productVector.size(); ++i){
+        cout << productVector.at(i);
+    }
+    
+    for (i = 0; i < descriptionVector.size(); ++i){
+        cout << descriptionVector.at(i);
+    }
+}
+
+void Product_and_Services::SaveToFile(vector<string>& productVector, vector<string>& descriptionVector){
```

Comment:
This one is similar to the ReadFromFile function.

This time we use `ofstream outputStreamVariable;` telling the program that we are using the variable outputStreamVariable to output information to the file.

To open the file we use `outputStreamVariable.open("TextFile.txt");` but we're still using the same text file as before, "SaleRepresentativesFile.txt".

Also, you should probably include the string variables product and serviceDescription in the function's arguments because we can't output to the file straight from the vector, at least it didn't work with me during my last project. So that would look like `product = productVector.at(i);` and then we start outputting information into the file using `outputStreamVariable << product;`. And that goes for the decriptionVector too.

After the loop is done we close the file with `outputStreamVariable.close();` and that's it, besides from making it look pretty in the file.